paper-input-container.html

```diff
@@ -365,10 +366,14 @@
       var value = inputElement[this._propertyForValue] || inputElement.value;
 
       var valid;
-      if (inputElement.validate) {
-        valid = inputElement.validate(value);
-      } else if (inputElement.checkValidity) {
-        valid = inputElement.checkValidity();
+
+      // type="number" doesn't have a validate method...
+      if (this.autoValidate) {
+        if (inputElement.validate) {
+          valid = inputElement.validate(value);
+        } else {
+          valid = inputElement.checkValidity();
+        }
       } else {
         valid = true;
```

Comment:
for `type="number"` it works by magic because the input itself doesn't let you type invalid input. For a regular input though, you're right, it sets it as invalid.

The problem is that I don't want to call `validate()` like we used to unless we need the value, since `validate()` now fires an event and will badge the decorator red if the input is invalid (which we don't need to do in the `auto-validate=false` case)